edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesRuntimeInfoProviderTest.cs

```diff
@@ -160,209 +164,198 @@ public async Task ConvertsPodsToModules()
-
-            edgehubTerminatedReason = "Completed";
-            modified["edgehub"].Status.Phase = "Pending";
-            modified["edgehub"].Status.ContainerStatuses[0].State.Running = null;
-            modified["edgehub"].Status.ContainerStatuses[0].State.Waiting = null;
-            modified["edgehub"].Status.ContainerStatuses[0].State.Terminated = new V1ContainerStateTerminated(0, finishedAt: DateTime.Parse("2019-06-12T16:13:07Z"), startedAt: DateTime.Parse("2019-06-12T16:11:22Z"), reason: edgehubTerminatedReason);
-
-            modified["simulatedtemperaturesensor"].Status.Phase = "Pending";
-            modified["simulatedtemperaturesensor"].Status.ContainerStatuses[1].State.Running = new V1ContainerStateRunning(startedAt: DateTime.Parse("2019-06-12T16:11:22Z"));
-            modified["simulatedtemperaturesensor"].Status.ContainerStatuses[1].State.Waiting = null;
-            modified["simulatedtemperaturesensor"].Status.ContainerStatuses[1].State.Terminated = null;
+            };
+            return pod;
+        }
 
-            foreach (V1Pod pod in modified.Values)
+        public static V1Pod CreatePodWithPodParametersOnly(string podPhase, string podReason, string podMessage)
+        {
+            V1Pod pod = new V1Pod
             {
-                runtimeInfo.CreateOrUpdateAddPodInfo(pod);
-            }
-
-            var pendingPhaseModules = (await runtimeInfo.GetModules(CancellationToken.None)).ToList();
-            Assert.Equal(3, pendingPhaseModules.Count);
+                Metadata = new V1ObjectMeta
+                {
+                    Name = "module-a-abc123",
+                    Labels = new Dictionary<string, string>
+                    {
+                        [KubernetesConstants.K8sEdgeModuleLabel] = "module-a"
+                    },
+                    Annotations = new Dictionary<string, string>
+                    {
+                        [KubernetesConstants.K8sEdgeOriginalModuleId] = "Module-A"
+                    }
+                },
+                Status = new V1PodStatus { Phase = podPhase, Reason = podReason, Message = podMessage },
+            };
+            return pod;
+        }
 
-            foreach (var i in pendingPhaseModules)
+        public static IEnumerable<object[]> GetListOfPodsInRunningPhase()
+        {
+            return new[]
             {
-                if (string.Equals("edgeAgent", i.Name))
+                new object[]
                 {
-                    Assert.Equal(ModuleStatus.Backoff, i.ModuleStatus);
-                    Assert.Equal($"Module in Back-off because of the reason: {agentWaitingReason}", i.Description);
-                }
-                else if (string.Equals("edgeHub", i.Name))
+                    CreatePodInPhaseWithContainerStatus("Running", new V1ContainerState(waiting: new V1ContainerStateWaiting("Waiting", "CrashBackLoopOff"))),
+                    $"Module in Back-off reason: CrashBackLoopOff",
```

Comment:
`$` means that the following string is [interpolated](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) one and it's not needed here and below.